src/runtime_src/core/tools/common/tests/TestHelper.h

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023-2024 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "tools/common/TestRunner.h"
+
+// Class representing a set of buffer objects (BOs)
+class BO_set {
+  xrt::bo bo_instr;   // Buffer object for instructions
+  xrt::bo bo_ifm;     // Buffer object for input feature map
+  xrt::bo bo_param;   // Buffer object for parameters
+  xrt::bo bo_ofm;     // Buffer object for output feature map
+  xrt::bo bo_inter;   // Buffer object for intermediate data
+  xrt::bo bo_mc;      // Buffer object for memory controller
+  uint32_t instr_size; // Size of the instruction buffer
+  size_t buffer_size; // Size of the buffer
+
+public:
+  // Constructor to initialize buffer objects
+  BO_set(xrt::device&, xrt::kernel&, size_t);
+
+  // Method to set kernel arguments
+  void set_kernel_args(xrt::run&);
+
+  // Method to synchronize buffer objects to the device
+  void sync_bos_to_device();
+};
+
+
+// Class representing a test case, which is created for a single run on a single thread//
+class TestCase {
+  xrt::device device;         // Device object
+  xrt::xclbin xclbin;         // Xclbin object
+  std::string kernel_name;    // Name of the kernel
+  xrt::hw_context hw_ctx;     // Hardware context
+  uint32_t queue_len = 4;     // Queue length
+  size_t buffer_size;         // Size of the buffer
+  int itr_count;              // Number of iterations
+
+  // Method to signal that a thread is ready to run
+  void thread_ready_to_run(std::mutex&, std::condition_variable&, uint32_t&);
+
+public:
+  // Constructor to initialize the test case with xclbin and kernel name with hardware context creation
+  TestCase(xrt::xclbin& xclbin, std::string& kernel, size_t buffer_size = 1024)
+      : device(xrt::device(0)), xclbin(xclbin), kernel_name(kernel), buffer_size(buffer_size), itr_count(1000) 
```

Comment:
Is it ok to blindly use device 0 for initialization? I know we are dealing with a single device right now, but it will be good to think of the pros/cons for the future

Reply:
Good catch. It should be the responsibility of individual test to set the device. This is now handled in the constructor itself.